### ISSUE TITLE
Outgoing connections not working after 2024.6 update

### ISSUE THREAD

**Issue by zedd3v** [mullvad/mullvadvpn-app#7058]
### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

Outgoing connections not working after 2024.6 update

The app also shows no outgoing ip
![CleanShot 2024-10-24 at 01 17 09](https://github.com/user-attachments/assets/15e67171-581a-4637-9001-4ac38989538e)

These are the only out of place logs I found, idk if related
[DEBUG] no one is waiting for ICMP packet (V4(Icmpv4Packet { ...
[DEBUG] error decoding ICMP packet: MalformedPacket(NotIcmpv4Packet)

### Expected Behavior

Expect it to work and show outgoing ip

### Steps to Reproduce

1. Install 2024.6
2. Connect

### Operating system version

macOS 15.0.1 (24A348)

### Mullvad VPN app version

2024.6

### Additional Information

Everything still functional on 2024.5 so its definitely 2024.6

**Comment by islamsaka** [comment]
I can confirm the issue for Macos 15.0.1 (24A348). After trying for OpenVPN/wireguard and sub-internal options, I couldn't fix the issue temporarily and was downgraded to 2024.5, and it seems to work fine.

**Comment by beesyrupfactory** [comment]
I'm having the same problem. I might need to revert back to 2024.5. Tailscale and my other VPN provider were working fine before on my mac, but now I keep getting dns-forward-failing and no-derp-connection errors.

**Comment by blue-devil** [comment]
I am here for this!
I have updated my macOS mullvad to 2024.6 and now I cannot browse any websites. Interestingly I can ping.
I have a linux machine with Mullvad 2024.5 and I have no problem with connection.

More: I have downloaded 2024.05 from github; uninstalled 2024.6 and installed 2024.5 on my macos and now I have internet back

**Comment by yww-sde** [comment]
Same here, uninstall & reinstall by Homebrew, full access granted, macOS 14.7, can connect to any server, has In IP, never get OUT IP for same reason, lead to all internet blocked.

**Comment by Mike-571** [comment]
I confirm with Sequoia 15.0.1 and Mullvad app 2024.6 : connectivity completely lost. I switched back to 2024.5.
Please, pull back macOS app 2024.6 and add a direct link to download 2024.5  for those who have applied the update.

**Comment by humblecoder** [comment]
> I confirm with Sequoia 15.0.1 and Mullvad app 2024.6 : connectivity completely lost. I switched back to 2024.5. Please, pull back macOS app 2024.6 and add a direct link to download 2024.5 for those who have applied the update.

Same.  MacOS 15.0.1.  Uninstalled 2024.6 and reinstalled 2024.5.  🫠

**Comment by dairymilkbatman** [comment]
Reverted back to wireguard-tools/mullvad to regain connectivity that was lost due to the app update 2024.6.

**Comment by MarkusPettersson98** [comment]
Hello everyone, thanks for reporting. We have been made aware of the issue with the 2024.6 release on macOS and we are investigating it.

**Comment by ldexterldesign** [comment]
Here we go again...

> Sorry if I've missed it here, if a Mullvad (MV) version exists where this issue doesn't then please let me/us know so I/we can rollback?:
> If yes then MV please enhance your testing before releases - issue is very 😞
> - https://github.com/mullvad/mullvadvpn-app/issues/6521#issuecomment-2395589050

Will be contacting support for a refund - unable to use my VPN properly for the past month

Regards

**Comment by raksooo** [comment]
Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg

Let us know how well it works for you!

**Comment by humblecoder** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

Seems to work for me -- in the 30 second test I gave it.  Loaded multiple browsers and ran a couple 3rd party "speed tests"

**Comment by MarkusPettersson98** [comment]
Thanks for the hasty feedback, it helps us a lot! 💛

**Comment by Mike-571** [comment]
It works here with Sequoia 15.0.1, Split tunneling (Safari can't be excluded, but it was already the case before)  and LittleSnitch 6.1.2. I tried multiple browsers and after reboot. Good job

**Comment by zedd3v** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

Works

**Comment by BionicBison05** [comment]
No more issues with my custom DNS setup after updating to the dev build.

**Comment by Yell0whead** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

Works, thank you.

**Comment by piru** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

This version fixes the issue for me, too.

**Comment by UndercoverGoose** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

Just checked this build out for myself, and I am still running into DNS problems _only_ when trying to run custom DNS outside of Mullvad. I am testing on Sequoia 15.1 (24B82). Specifically, trying to run Cloudflare WARP through their desktop application alongside Mullvad breaks all DNS resolution. However, defining a custom DNS within Mullvad does work properly. AFAIK everything else works, including split tunneling and Win11 Parallels (someone mentioned it not working above).

**Comment by nicolasharnois** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

Seems to fix the issue on macOS 15.0.1 for me as well. I have another computer running macOS 15.0 (not yet upgraded to 15.0.1) with Mullvad 2024.6 and there are no issues on that one.

**Comment by nicolasharnois** [comment]
> This works. To note, Windows 11 running via Parallels encounters issues. States no Internet connection in Taskbar tray, and internet doesn't connect reliably in browser.

I'm seeing this as well. I switched to Bridge Network for the time being and it works.

**Comment by Artimunor** [comment]
> raksooo

this works however it shows a security warning: "Your privacy might be at risk with this unsupported app version. Please update now."

**Comment by nijaru** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

Working here.

**Comment by voltzd** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

I was just sent this version via email by support, has fixed the issue for me too but I needed to update my macOS or restart for it to work, not sure which one as I just updated.

**Comment by nicolasembleton** [comment]
I've just tried on MacOS Sonoma 14.7 (23H124) with the latest 2024.7-beta1 and the issue seems to be the same than 2024.6. Ingoing is ok but outgoing never gets filled. I get lots of really weird behaviors on the UI such as impossible to switch to another location, switches seem to not be visually responding, like the UI thread was overloaded with something and never gotten CPU time back. Pretty strange. Had to downgrade to 2024.5 for it to work. Behavior when connecting is quite different on 2024.5 than 2024.6+, whereby "creating quantum connection" takes a good 10 seconds before the "out" field gets filled, whereas on 2024.6+, it is very quick, but it never gets filled. Not sure whether this extra info helps or not.

**Comment by raksooo** [comment]
We've now released 2024.7-beta1 which contains the same fix as in the dev-build posted above. Please try out the beta! Thanks to everyone who helped test the dev-build! If you run into any issues on future betas, please report the issue either with the "Report a problem" function in the app or here.

@nicolasembleton Could you send a problem report in the app and write that it is in response to this issue, and we'll take a look!

**Comment by nicolasharnois** [comment]
> We've now released 2024.7-beta1 which contains the same fix as in the dev-build posted above. Please try out the beta! 

Looks good at my end, thanks @raksooo. There's still the issue of Shared Network no longer working in Parallels (Win 11 VM, Edge returns DNS_PROBE_FINISHED_NO_INTERNET).

**Comment by beesyrupfactory** [comment]
When using 2024.5 release, I was able to run Tailscale as an exit node on my computer using Mullvad VPN.

I just tried to run Tailscale with 2024.7-beta1, and I'm still seeing dns-forward-failing on the Tailscale side. As this is the case, I will need to revert back to 2024.5 again.

**Comment by ldexterldesign** [comment]
macos 15.1 (24B82)
mullvad 2024.7-beta1

AirDrop (still) broken

Regards

**Comment by Artimunor** [comment]
> macos 15.1 (24B82) mullvad 2024.7-beta1
> 
> AirDrop (still) broken
> 
> Regards

i can confirm that in the 2024.6-devbuild airdrop works

**Comment by vanyauhalin** [comment]
Perhaps my issue is related to this one. After the [update], when starting Docker services, any connection to the Internet stops functioning. This was reproduced when working with Docker versions [4.29.0] and [4.35.0] as well. I reverted to Mullvad version [2024.5], and this solved my problem.

[update]: https://github.com/Homebrew/homebrew-cask/blob/c43605a3e33a7333ac96cabec244321b7a982c3f/Casks/m/mullvadvpn.rb
[2024.5]: https://github.com/Homebrew/homebrew-cask/blob/da5a0d051e919f85f1c7bec764f01ca863b01d9c/Casks/m/mullvadvpn.rb
[4.35.0]: https://github.com/Homebrew/homebrew-cask/blob/60d0e0fe5a63d4ef8871fceeafb6a8750c925d5c/Casks/d/docker.rb
[4.29.0]: https://github.com/Homebrew/homebrew-cask/blob/cadb3aad3e1f57386fc7d2173b76feafcf1a4302/Casks/d/docker.rb

**Comment by AlecGower** [comment]
> Hi! We have a potential fix that we would appreciate if you all could help us try out! It's somewhat difficult to reproduce so we can't reliably test it on our computers. Here's a test build that includes the fix: https://releases.mullvad.net/desktop/builds/2024.6-dev-8e55ad%2Btest-20247/MullvadVPN-2024.6-dev-8e55ad%2Btest-20247.pkg
> 
> Let us know how well it works for you!

This fix worked for me as well. For the time being, until this is incorporated in a new release, should I revert back to 2024.5? I got a warning message in the app saying this test version is unsupported.

**Comment by MarkusPettersson98** [comment]
> This fix worked for me as well (M1 Pro Sonoma 14.7). For the time being, until this is incorporated in a new release, should I revert back to 2024.5? I got a warning message in the app saying this test version is unsupported.

The fix has been released in the current beta: https://github.com/mullvad/mullvadvpn-app/releases/tag/2024.7-beta1
We are looking at releasing a stable version asap, but in the meantime you could install this beta release. It will not complain about being "unsupported" :blush:

**Comment by ldexterldesign** [comment]
> macos 15.1 (24B82) mullvad 2024.7-beta1
> 
> AirDrop (still) broken
> 
> Regards

Please ignore this

I neglected to turn on `local network sharing` after a fresh install of your beta

Apologies

**Comment by MarkusPettersson98** [comment]
2024.7 has been released as stable: https://github.com/mullvad/mullvadvpn-app/releases/tag/2024.7 :rocket:

**Comment by vanyauhalin** [comment]
> Perhaps my issue is related to this one. After the [update](https://github.com/Homebrew/homebrew-cask/blob/c43605a3e33a7333ac96cabec244321b7a982c3f/Casks/m/mullvadvpn.rb), when starting Docker services, any connection to the Internet stops functioning. This was reproduced when working with Docker versions [4.29.0](https://github.com/Homebrew/homebrew-cask/blob/cadb3aad3e1f57386fc7d2173b76feafcf1a4302/Casks/d/docker.rb) and [4.35.0](https://github.com/Homebrew/homebrew-cask/blob/60d0e0fe5a63d4ef8871fceeafb6a8750c925d5c/Casks/d/docker.rb) as well. I reverted to Mullvad version [2024.5](https://github.com/Homebrew/homebrew-cask/blob/da5a0d051e919f85f1c7bec764f01ca863b01d9c/Casks/m/mullvadvpn.rb), and this solved my problem.

I have just updated to [2024.7]. At first glance, my problem appears to be solved. Thank you, @MarkusPettersson98, for such a quick fix. ❤️

[2024.7]: https://github.com/Homebrew/homebrew-cask/blob/d61299c7a04e9abb2b25a37e52e7a4c25c1d2b73/Casks/m/mullvadvpn.rb

**Comment by nicolasembleton** [comment]
> We've now released 2024.7-beta1 which contains the same fix as in the dev-build posted above. Please try out the beta! Thanks to everyone who helped test the dev-build! If you run into any issues on future betas, please report the issue either with the "Report a problem" function in the app or here.
> 
> @nicolasembleton Could you send a problem report in the app and write that it is in response to this issue, and we'll take a look!

Sorry for the late reply. I just did and mentioned the symptoms (but forgot to mention this particular issue#). I tried the latest stable version and I'm still having the same issue on Sonoma.

**Comment by Mike-571** [comment]
Problem solved. Thanks for your great support.

**Comment by islamsaka** [comment]
> I can confirm the issue for Macos 15.0.1 (24A348). After trying for OpenVPN/wireguard and sub-internal options, I couldn't fix the issue temporarily and was downgraded to 2024.5, and it seems to work fine.

I've tested the 2024.7 release on the same device, and everything is fine. Thanks for the quick fix.

**Comment by beesyrupfactory** [comment]
My apologies, but I've just tested the final release of 2024.7, and I'm still having issues while running a Tailscale exit node on my Mac while running Mullvad VPN. If this requires a new request, please let me know and I will be happy to share steps to reproduce this behavior. This does not happen on the 2024.5 release of Mullvad VPN. The error on the Tailscale client side is `dns-forward-failing`. I know this has nothing to do with Mullvad, but the fact that Tailscale exit node is still working when I revert to 2024.5 means that there is some change that I need to account for on the Mullvad side.

**Comment by floq-design** [comment]
2024.7 still not working for me with local development setup. Using Valet and TLD .test.
Reverting back to 2024.5 works fine.

**Comment by jmonday** [comment]
2024.7 is not working for me either. I'm using dnsmasq with local web development. When connected to the VPN, my local development environment doesn't respond as expected. I see the "this site can't be reached" with a `DNS_PROBE_POSSIBLE` error.

After downgrading to 2024.5, my local development site responds as expected.

**Comment by Mike-571** [comment]
Using manual DNS within host, as indicated on this page : https://mullvad.net/en/help/dns-over-https-and-dns-over-tls#win11 But it's a lot slower than automatic DNS.

**Comment by beesyrupfactory** [comment]
Hello, I apologize for the bump, but I wanted to determine if there were any progress with this case.

My specific use case involves tailscale client running concurrently with Mullvad where the machine serves as an exit node. This worked previous in 2024.5, but after the past two revisions, I am still stuck on 2024.5 at the moment. Please let me know if you require any diagnostics.

**Comment by MarkusPettersson98** [comment]
Hello, thanks for the report! We are aware that some users experience issues with using Mullvad and Tailscale on macos with post-2024.5 releases. Unfortunately, I can not give you any more than that at the moment, but we'll make sure to keep you guys updated :blush:

**Comment by humblecoder** [comment]
Well, here's an interesting (¿edge?) case.

`MacOS Host (Mullvad 2024.7) -> Parallels (Debian 12) -> NO INTERNET`

`MacOS Host (Mullvad 2024.7) -> Parallels (Debian 12) (guest also running Mullvad) -> SUCCESSFUL INTERNET`

Both are set to "Shared Network" for the guests.

😳

**Comment by MarkusPettersson98** [comment]
Whoops, did not mean to close this issue. Let's reopen it for now. 

But I do have some good news, we finally have some updates on this issue! :tada: 

Go check out the PR https://github.com/mullvad/mullvadvpn-app/pull/7256 if you would like to know more about the technical details.
TL;DR is that Orbstack and (potentially other apps) would mess with our use of `pf` and local DNS resolver since app version 2024.6. Users on macOS versions prior to 14.6 _or_ after 15.1 will be reverted to the behavior present in the 2024.5 release and prior.

We are planning to make a beta release with this fix included very soon, so stay tuned. Thanks for all of your reports! :yellow_heart:

**Comment by beesyrupfactory** [comment]
Hi @MarkusPettersson98 ,

Happy new year. I wanted to check in and see if this fix would be available in the latest 2025.1 release. I see that #7256 is merged, but as this ticket is still open, I wanted your confirmation before I proceed to install 2025.1.

Thanks!

**Comment by beesyrupfactory** [comment]
Hi @MarkusPettersson98 ,

Thanks so much for your response. I'm no longer seeing issues with connectivity when using Tailscale and the Mullvad VPN app.

Appreciate your support!

**Comment by MarkusPettersson98** [comment]
That's great, thanks for the quick reply! :yellow_heart: 

Keeping this open until we have rolled out the new release completely :blush:

**Comment by nijaru** [comment]
> Hi @beesyrupfactory,
> 
> Happy new year! We have started to gradually roll out 2025.1 at this point, so please go ahead and install it! You'll find the macOS release here on GitHub for now, but it should be available on the website in the coming days 😊
> 
> I'm keeping this issue open until we receive more feedback on the fix, that's all 😄

Is there no beta version currently? I've been running a beta version that was older than 2025.1 and didn't see any notification that 2025.1 was available.

**Comment by beesyrupfactory** [comment]
@nijaru I've been maintaining my mullvad app installation with homebrew cask, so that's how I became aware of the 2025.1 update.

**Comment by nijaru** [comment]
> @nijaru I've been maintaining my mullvad app installation with homebrew cask, so that's how I became aware of the 2025.1 update.

I've been doing the same, but the `mullvadvpn@beta` cask is at version `2024.9-beta1` whereas the regular `mullvadvpn` cask is at `2025.1`. I just installed from the stable cask for now, but I did want to bring it up.

I may have just missed the notification, but I did want to double check that as I don't see any newer beta version currently.

**Comment by MarkusPettersson98** [comment]
We are slowly rolling out this release due to holidays affecting staff availability, so the in-app notification for the 2025.1 release has not been pushed out yet. I hope that clears up the situation :blush:

**Comment by MarkusPettersson98** [comment]
Forgot to close this issue, but it seems to have been resolved. Thanks yet again for reporting it to us!
Stay safe 💛
